### PR TITLE
The JWS installation option should allow you to exclude natives

### LINKED ISSUE #73: The JWS installation option should allow you to exclude natives
##### SUMMARY
Not everyone wants to deploy the natives zip in their environment, so we should easily allow you to disable that requirement. I don't see an easy way to do so in the current setup. I've tried setting the `jws_native_zipfile_url` to an empty string but it still tries.

```
TASK [middleware_automation.jws.jws : Install JWS native dependencies with zipfile from RHN:] ********************************************************************************************************************************************************************************************************************************
skipping: [localhost]

TASK [middleware_automation.jws.jws : Copy archives to target nodes] *********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => (item=jws-5.6.0-application-server.zip)
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: If you are using a module and expect the file to exist on the remote, see the remote_src option
failed: [localhost] (item=jws-5.6.0-application-server-RHEL35-x86_64.zip) => {"ansible_loop_var": "item", "changed": false, "item": "jws-5.6.0-application-server-RHEL35-x86_64.zip", "msg": "Could not find or access '/root/jws-5.6.0-application-server-RHEL35-x86_64.zip' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}
```

Note that I'm testing on Fedora and even though it's an unsupported configuration I expected the JWS java only zip to install and run correctly.

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by rpelisse**
Fixed #73 (and reduce code duplication)

### CODE REVIEW

**Comment by rpelisse** [comment]
@guidograzioli Can you review and approve? 
 @csutherl can you test to be sure that native is no longer an obligation?

**Comment by guidograzioli** [comment]
I think this PR contains many same commits as #89? Can you check?

**Comment by rpelisse** [comment]
@guidograzioli yes, it's designed to be merge after PR89.

**Comment by guidograzioli** [comment]
LGTM!

**Review by guidograzioli — APPROVED** [review]

File: roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    path: "{{ path_to_zipfile }}"
+  register: archive_path
+
+- name: Check local download archive path
+  ansible.builtin.stat:
+    path: "{{ lookup('env', 'PWD') }}"
+  register: local_path
+  delegate_to: localhost
+
+- name: "Check downloaded archive local_path.stat.path }}/{{ zipfile }}"
```
Comment (csutherl) [inline roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml:23]: You're missing the opening curly braces for the variable here.

**Comment by csutherl** [comment]
I cloned the PR branch and tested it a bit, and found that there's one place you didn't add a `tomcat_native` check which was causing the native zipfile to still be required. Adding a check for `tomcat_native == True` to the `when` block in [roles/jws/tasks/install/zipfiles.yml#L16](https://github.com/ansible-middleware/jws-ansible-playbook/blob/main/roles/jws/tasks/install/zipfiles.yml#L16) makes it work as expected.

I did notice another thing that we may want to address. If you run the script with the default `tomcat_native: False` and then run it again after setting `tomcat_native: True` it unpacks the zip as expected, but doesn't notify a restart of the Tomcat service. I had to manually restart the process for tomcat_native to be picked up. If we think this behavior is ok, we should probably document that a restart is required (after someone checks to make sure it's not an env problem).

**Comment by rpelisse** [comment]
@csutherl you are a god send! Thank you for spotting all of that. I think I've fixed everything and this PR should be good to go. However, if you can review (and /or test it) one last time, that certainly can't hurt :)

**Comment by csutherl** [comment]
@rpelisse I have a particular set of skills (spotting typos) :laughing: My day is kind of full today, but I should be able to give it one more go this afternoon and provide some feedback. Thanks for being so quick to respond!

File: roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml (hunk 1 above)
Comment (csutherl) [inline roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml:23]: The opening braces are still missing :(

**Review by csutherl — CHANGES_REQUESTED** [review]
I just tested the latest commit and it still doesn't work unless you add the `tomcat_native` check to `roles/jws/tasks/install/zipfiles.yml` so that the `zipfiles` list doesn't require the libtcnative.so. Without that check in place, you see the following failure when you run the playbook (having `jws_version` and `tomcat_install_method` defined):

```
TASK [middleware_automation.jws.jws : Install Jboss Web Server and required binaries from local zipfiles (install method: rhn_zipfiles] **************************************************************************************************************************************************************************************
changed: [localhost] => (item={'src': 'jws-5.6.0-application-server.zip', 'creates': '/opt/jws-5.6/tomcat/bin'})
failed: [localhost] (item={'src': 'jws-5.6.0-application-server-RHEL35-x86_64.zip', 'creates': '/opt/jws-5.6/tomcat/lib/libtcnative-1.so'}) => {"ansible_loop_var": "item", "changed": false, "item": {"creates": "/opt/jws-5.6/tomcat/lib/libtcnative-1.so", "src": "jws-5.6.0-application-server-RHEL35-x86_64.zip"}, "msg": "Source '/opt/jws-5.6.0-application-server-RHEL35-x86_64.zip' does not exist"}
```

File: roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml (hunk 1 above)
Comment (rpelisse) [reply roles/jws/tasks/install/fetch_zipfile_from_rhn_and_deploy.yml:23]: Darn! This was the very first thing I fixed up! It must have been squashed when I rebase :(

**Comment by rpelisse** [comment]
@csutherl OK, confirmed... Rebasing has removed all my changes... I guess "Don't rebase without testing". I'm going to update the PR. Sorry for that :(

**Comment by rpelisse** [comment]
@csutherl OK, added back the change, tested locally. But if you have the time to review again, that'll be awesome.

**Review by csutherl — APPROVED** [review]
LGTM! I appreciate you adding the restart notification, that works well too.

**Comment by rpelisse** [comment]
@csutherl Cool! No problem, it was a good idea to add notify :) Then I'll merge this right away :)

**Pull request merged**